File: OSX/Feet-Inches-Cm using PickerView/Feet-Inches-Cm using PickerView/ViewController.m
//
//  ViewController.m
//  Feet-inches-cm converter
//
//  Created by Adhitya Surya Pratama on 10/30/16.
//  Copyright © 2016 Adhitya Surya Pratama. All rights reserved.
//

#import "ViewController.h"

@interface ViewController (){
    NSArray *_pickerData;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _pickerData = @[@"Select",@"Inch-Centimeters",@"Centimeters-Inch"];
    
    // Do any additional setup after loading the view, typically from a nib.
    self.picker.delegate = self;
    self.picker.dataSource = self;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView{
    return 1;
}


- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component{
    return _pickerData.count;
    
}


- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    return _pickerData[row];
}


- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    if(row == 1){
        _resultHeight.text =[NSString stringWithFormat:@"%f",_originalHeight.text.doubleValue * 2.54];
    }
    else if(row == 2){
        _resultHeight.text = [NSString stringWithFormat:@"%f",_originalHeight.text.doubleValue /2.54];
    }
}


@end

File: DraggableButton/DraggableButton/ViewController.swift
//
//  ViewController.swift
//  DraggableButton
//
//  Created by Adhitya Surya Pratama on 11/8/17.
//  Copyright © 2017 Adhitya Surya Pratama. All rights reserved.
//

import UIKit
import DragDropiOS

class ViewController: UIViewController {
    
    @IBOutlet weak var button : UIButton!

    @IBOutlet weak var dragDropView: DroppableView!
    @IBOutlet weak var collectionView: UICollectionView!
    @IBOutlet weak var tableView: UITableView!
    var dragDropManager : DragDropManager!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        button.center = self.view.center
        button.layer.cornerRadius = 3
        button.layer.masksToBounds = true
        button.backgroundColor = UIColor.yellow
        button.setTitle("NORMAL", for: .normal)
        button.setTitle("Floating", for: .highlighted)
        button.addTarget(self,
                         action: #selector(drag(control:event:)),
                         for: UIControlEvents.touchDragInside)
        button.addTarget(self,
                         action: #selector(drag(control:event:)),
                         for: [UIControlEvents.touchDragExit,
                               UIControlEvents.touchDragOutside])
        dragDropManager = DragDropManager(canvas: self.view, views: [dragDropView])

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @objc func drag(control: UIControl, event: UIEvent) {
        if let center = event.allTouches?.first?.location(in: self.view) {
            control.center = center
        }
    }
}

